backend/src/database/state-migrations/8-treatment-system-improvements.ts

```diff
@@ -0,0 +1,118 @@
+import type {
+    ExerciseState,
+    ImageProperties,
+    Mutable,
+} from 'digital-fuesim-manv-shared';
+import {
+    cloneDeepMutable,
+    SpatialTree,
+    StrictObject,
+} from 'digital-fuesim-manv-shared';
+import type { Migration } from './migrations';
+
+export const treatmentSystemImprovements8: Migration = {
+    actions: (_initialState, actions: any[]) => {
+        for (const action of actions) {
+            switch (action?.type) {
+                case '[Vehicle] Add vehicle':
+                    for (const material of action.materials) {
+                        migrateMaterial(material);
+                    }
+                    for (const personnel of action.personnel) {
+                        migratePersonnel(personnel);
+                    }
+                    break;
+                case '[Patient] Add patient':
+                    migratePatient(action.patient);
+                    break;
+            }
+        }
+    },
+    state: (state: any) => {
+        // Spatial tree
+        state.spatialTrees = cloneDeepMutable({
+            materials: SpatialTree.create(),
+            patients: SpatialTree.create(),
+            personnel: SpatialTree.create(),
+        });
+        for (const material of StrictObject.values(state.materials)) {
```

Comment:
Any particular reason for `StrictObject` over `Object`?

Reply:
I believe `StrictObject` should be the default to use in this codebase. It is correct that the stricter typings don't provide a benefit in this case.

Reply:
Sounds reasonable, but we should document this somewhere.

Reply:
Added in 8e3439e5b9623ae14530cfd879e8a4a6bd11cc2d